lib/workers/repository/onboarding/branch/index.ts

```diff
@@ -47,28 +50,16 @@ export async function checkOnboardingBranch(
   setGitAuthor(config.gitAuthor);
   const onboardingPr = await getOnboardingPr(config);
   if (onboardingPr) {
+    logger.debug('Onboarding PR already exists');
     if (config.onboardingRebaseCheckbox) {
       handleOnboardingManualRebase(onboardingPr);
     }
-    logger.debug('Onboarding PR already exists');
-    const { rawConfigHash } = onboardingPr.bodyStruct ?? {};
-    const commit = await rebaseOnboardingBranch(config, rawConfigHash);
-    if (commit) {
-      logger.info(
-        { branch: config.onboardingBranch, commit, onboarding: true },
-        'Branch updated'
+    if (await isOnboardingBranchModified(config.onboardingBranch!)) {
+      invalidateExtractCache(config.baseBranch!);
```

Comment:
This seems wrong. If the onboarding branch has been modified at any time at all (which is common, whenever users change their config), then we always invalidate the extract cache?

Reply:
It will behave correctly actually. The fn `isOnboardingBranchModified`  checks if the branch has been modified relative to last run rather than just check if its modified by user or not. I will add documentation for this.

Reply:
hasOnboardingBranchChanged()